base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -261,7 +267,29 @@ impl LMDBDatabase {
                     height,
                     hash,
                     accumulated_difficulty,
+                    from,
                 } => {
+                    // for security we check that the best block does exist, and we check the previous value
+                    // we dont want to check this if the prev block has never been set, this means a empty hash of 32
+                    // bytes.
+                    if from != vec![0; 32] {
```

Comment:
1. AFAICS you want to ignore the genesis block - so can use `if height > 0`?
2. `from` is a bit fuzzy, perhaps `expected_prev_block`?